File: Preproc/create_tsv_files.pl
#!/usr/bin/perl -w

use strict;
use vars qw ();

use List::Util qw/shuffle/;

my $data_d = $ARGV[0];
my $split_d = $ARGV[1];
my $out_d = $ARGV[2];

if (not -e $out_d) {
    system("mkdir $out_d");}

opendir(D,"$split_d")||die;
my @File = grep /\.tsv/,readdir D;
closedir(D);
    
foreach my $f (@File) {
    open(I,"$split_d/$f")||die;
    open(O,">$out_d/$f")||die;
    while(<I>) {
	chomp;
	my $f2 = $_;

	if ($f =~ /^train/) {
	    open(I2,"$data_d/Train/$f2")||die "$data_d/Train/$f2\n";}
	else {
	    open(I2,"$data_d/Test/$f2")||die;}
	while(<I2>) {
	    print O "$_";
	}
	close(I2);
	print O "\n";
    }
    close(O);
    close(I);
}

File: Preproc_py/create_tsv_files.py
import sys
import glob
import os

data_d = sys.argv[1]
split_d = sys.argv[2]
out_d = data_d + "/conll"

if not os.path.isdir(out_d):
    os.mkdir(out_d)

os.chdir(split_d)
Files = glob.glob("*")

for file in Files:
    dir_f = open(split_d+"/"+file,'r')
    out_f = open(out_d+"/"+file,'w')

    for l in dir_f:
        l = l.rstrip()

        if 'train' in file:
            in_f = open(data_d+"/Train/"+l,'r')
        else:
            in_f = open(data_d+"/Test/"+l,'r')

        for line in in_f:
            out_f.write(line)
        in_f.close()

    dir_f.close()
    out_f.close()
